### PR TITLE
fix(ui): Convert truncate to support dark mode

### PR DESCRIPTION
**Pull request by Zylphrex**
The truncate component is currently styled in the less files. This means that it
is not respecting dark themes. This change moves the styling to use emotion so
it follows the dark theme colors.

# Before

https://user-images.githubusercontent.com/10239353/110511759-91a61400-80d2-11eb-922e-181d2c4c7b93.mp4

# After

https://user-images.githubusercontent.com/10239353/110511783-97035e80-80d2-11eb-8800-cfb9755f3327.mp4

### CODE REVIEW

File: src/sentry/static/sentry/less/shared-components.less [hunk 1]
```diff
@@ -1484,28 +1484,6 @@ ul.radio-inputs {
   }
 }
 
-/**
- * Truncate component.
- */
-.truncated {
```
Comment (billyvg) [inline src/sentry/static/sentry/less/shared-components.less:1490]: Did you check if we use this in `getsentry`?

Comment (Zylphrex) [reply src/sentry/static/sentry/less/shared-components.less:1490]: Can confirm that `getsentry` does not use this anywhere.

File: src/sentry/static/sentry/app/components/truncate.tsx [hunk 2]
```diff
@@ -75,25 +87,52 @@ class Truncate extends React.Component<Props, State> {
...
+  white-space: nowrap;
+  border-radius: ${space(0.5)};
+  top: -5px;
+  ${p => p.expandDirection === 'left' && 'right: -5px;'}
+  ${p => p.expandDirection === 'right' && 'left: -5px;'}
+
+  ${p =>
+    p.expanded &&
+    `
+    z-index: 10;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/truncate.tsx:133]: Can we add this to theme? there's a `zIndex` section

**Review by priscilawebdev — COMMENTED** [review]
the code looks good to me, but it would be nice to see the before and after changes. 

I noticed that this component is not part of the storybook. Can we maybe add it there?

**Review by billyvg — APPROVED** [review]

**Review by priscilawebdev — APPROVED** [review]
Thank you for adding the before/after record and also for adding the component to the Storybook. Great Work! 👏

**Pull request merged**